src/eu/deltasource/internship/ecosystem/ecosystems/Ecosystem.java

```diff
@@ -0,0 +1,184 @@
+
+    public static void init(List<Animal> animalCollection) {
+
+        for (int i = 0; i < animalCollection.size(); i++) {
+            if (animalCollection.get(i) instanceof Carnivore) {
+                carnivores.add((Carnivore) animalCollection.get(i));
+            } else {
+                herbivores.add((Herbivore) animalCollection.get(i));
+            }
+        }
+    }
+
+    public static void run(int iterations) {
+
+        reproduce();
+        attack();
+        age();
+        ageCheck();
+        increaseStarvation();
+        starvationCheck();
+        cleanDeadAnimals();
+        sortLists();
+        System.out.println("Iteration : " + iterations);
+        System.out.println("Herbivores : " + herbivores.size());
+        System.out.println("Carnivore : " + carnivores.size() + "\n");
+
+
+    }
+
+    /**
+     * Sorts the list of carnivores and herbivores.
+     */
+    private static void sortLists() {
+        Collections.sort(carnivores);
+        Collections.sort(herbivores);
+    }
+
+    /**
+     * Increase the hunger level of the carnivores by their hunger rate modifier
+     */
+    private static void increaseStarvation() {
+        for (int i = 0; i < carnivores.size(); i++) {
+            carnivores.get(i).setHungerLevel(carnivores.get(i).getHungerLevel() + carnivores.get(i)
+                    .getHungerLevelModifier());
+        }
+    }
+
+    /**
+     * Carnivore attack a random herbivore and if it is successful check if the carnivore is a group animal or a
+     * lone animal and after that proceed eating it with the other animals or alone.
+     */
+    private static void attack() {
+        RandomNumberGeneratorRealInput randomNum = new RandomNumberGeneratorRealInput();
+        boolean isAttackSuccessful = false;
+        for (int i = 0; i < carnivores.size(); i++) {
+            if (herbivores.size() > 0) {
+                int chosenHerbivore = randomNum.randomNumber(0, herbivores.size() - 1);
+                isAttackSuccessful = carnivores.get(i).attack(herbivores.get(chosenHerbivore));
+                if (isAttackSuccessful == true) {
+                    System.out.println(carnivores.get(i).getClass().getName().substring(44) + " killed a "
+                    + herbivores.get(chosenHerbivore).getClass().getName().substring(44));
+                    if (carnivores.get(i).getLivingType().equals(LivingTypes.GROUP)){
+                        groupCarnivoreAttack(i, chosenHerbivore);
+                    } else {
+                        carnivores.get(i).eat(herbivores.get(chosenHerbivore), 1);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks how many animals are in the group, split the prey equally to them and after that they eat it.
+     * @param i takes the carnivore index.
+     * @param chosenHerbivore takes the herbivore that is attacked.
+     */
+    private static void groupCarnivoreAttack(int i, int chosenHerbivore) {
```

Comment:
Is it a group attack? I think the attack is already performed. (maybe think of a better name)
int i -> not a good param name